### PR TITLE
GA migration Precommit Java, multiple IOs

### PR DESCRIPTION
**Pull request by benWize**
As part of the migration of Precommit and Postcommit Jobs from Jenkins to GA in self-hosted runners, this PR contains:

- Migrated workflow [job-precommit-java-debezium-io-direct.yml](https://github.com/benWize/beam/blob/1f0af1afc7833f5812bc08f16b1d1660015f317e/.github/workflows/job-precommit-java-debezium-io-direct.yml)
- Migrated workflow [job-precommit-java-jdbc-io-direct.yml](https://github.com/benWize/beam/blob/1f0af1afc7833f5812bc08f16b1d1660015f317e/.github/workflows/job-precommit-java-jdbc-io-direct.yml)
- Migrated workflow [job-precommit-java-jms-io-direct.yml](https://github.com/benWize/beam/blob/1f0af1afc7833f5812bc08f16b1d1660015f317e/.github/workflows/job-precommit-java-jms-io-direct.yml)
- Migrated workflow [job-precommit-java-kafka-io-direct.yml](https://github.com/benWize/beam/blob/1f0af1afc7833f5812bc08f16b1d1660015f317e/.github/workflows/job-precommit-java-kafka-io-direct.yml)
- Migrated workflow [job-precommit-java-kinesis-io-direct.yml](https://github.com/benWize/beam/blob/1f0af1afc7833f5812bc08f16b1d1660015f317e/.github/workflows/job-precommit-java-kinesis-io-direct.yml)
- Migrated workflow [job-precommit-java-kudu-io-direct.yml](https://github.com/benWize/beam/blob/1f0af1afc7833f5812bc08f16b1d1660015f317e/.github/workflows/job-precommit-java-kudu-io-direct.yml)
- Migrated workflow [job-precommit-java-mongodb-io-direct.yml](https://github.com/benWize/beam/blob/1f0af1afc7833f5812bc08f16b1d1660015f317e/.github/workflows/job-precommit-java-mongodb-io-direct.yml)
- Migrated workflow [job-precommit-java-neo4j-io-direct.yml](https://github.com/benWize/beam/blob/1f0af1afc7833f5812bc08f16b1d1660015f317e/.github/workflows/job-precommit-java-neo4j-io-direct.yml)

The migrated workflows were removed from [.test-infra/jenkins/README.md](https://github.com/benWize/beam/blob/1f0af1afc7833f5812bc08f16b1d1660015f317e/.test-infra/jenkins/README.md) and added to [CI.md](https://github.com/benWize/beam/blob/1f0af1afc7833f5812bc08f16b1d1660015f317e/CI.md)

**DO NOT MERGE** until the effort to use **self-hosted runners** is completed https://github.com/apache/beam/pull/22703

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by benWize** [comment]
Successful executions on self-hosted runners:

- MongoDB https://github.com/benWize/beam/actions/runs/3018375043
- Neo4j https://github.com/benWize/beam/actions/runs/3018375041
- JMS https://github.com/benWize/beam/actions/runs/3018375040
- Debezium https://github.com/benWize/beam/actions/runs/3018375039
- JDBC https://github.com/benWize/beam/actions/runs/3018375045
- Kinesis https://github.com/benWize/beam/actions/runs/3018375047

**Comment by fernando-wizeline** [comment]
LGTM!

File: CI.md [hunk 1]
```diff
@@ -125,6 +125,56 @@ Service Account shall have following permissions ([IAM roles](https://cloud.goog
 | Java Wordcount Direct Runner | Runs Java WordCount example with Direct Runner.                                               | Yes              | Yes                   | Yes           | -                        |
 | Java Wordcount Dataflow      | Runs Java WordCount example with DataFlow Runner.                                             | -                | Yes                   | Yes           | Yes                      |
 
+### PreCommit Workflows
+
+#### Java Debezium IO Direct - [job-precommit-java-debezium-io-direct.yml](.github/workflows/job-precommit-java-debezium-io-direct.yml)
```
Comment (damccorm) [inline CI.md:130]: Rather than having a bunch of distinct sections can we just make one larger table? I don't think each of these sections add much

**Review by damccorm — COMMENTED** [review]
Had one small comment, otherwise LGTM

File: CI.md (hunk 1 above)
Comment (damccorm) [reply CI.md:130]: Also, this applies to the other similar PRs recently added.

Comment (benWize) [reply CI.md]: I changed it, thanks!

**Review by damccorm — APPROVED** [review]

**Review by Abacn — APPROVED** [review]
changes since last review LGTM. The CI.md conflicts should be trivial to resolve.

**Pull request closed without merging**